component/http/middleware.go

```diff
@@ -166,8 +242,20 @@ func NewCompressionMiddleware(deflateLevel int, ignoreRoutes ...string) Middlewa
 				log.Debugf("url %s skipped from compression middleware", r.URL.String())
 				return
 			}
-			// explicitly specify encoding in header
-			w.Header().Set(encoding.ContentEncodingHeader, hdr)
+
+			selectedEncoding, err := parseAcceptEncoding(hdr)
+			if err != nil {
+				// todo if a selectedEncoding is now known by a service, it should reply 406 Not Acceptable
```

Comment:
TODO???

Reply:
The proper decision of whether to respond 406 might be better done somewhere else. This middleware might be or might not be a good place for it. Ideally, there should be some kind of orchestrator that manages such "global" decisions for request.

This decision is definitely out of the scope of compression middleware change only. I believe it's better to have this [changed incrementally](https://github.com/google/eng-practices/blob/master/review/developer/small-cls.md#what-is-small-what_is_small) and not inflate the current scope.

A practice that I am used to in order to postpone such decisions, is creating a todo and opening an issue after the change has been merged. Without a todo that you can refer to in that issue, this might get lost.

How do you recommend doing this?

Reply:
Done https://github.com/beatlabs/patron/pull/305/commits/e0722420e9b3ecd242792f76bb5466eb2bcb4147